test/apps/mapbox-tile/utils/decode.js

```diff
@@ -0,0 +1,41 @@
+import Protobuf from 'pbf';
+import {VectorTile} from '@mapbox/vector-tile';
+import {worldToLngLat} from 'viewport-mercator-project';
+import {vectorTileFeatureToGeoJSON} from './feature';
+
+const TILE_SIZE = 512;
+
+export function decodeTiles(x, y, z, arrayBuff) {
+  const tile = new VectorTile(new Protobuf(arrayBuff));
+
+  const result = [];
+  const projectFunc = project.bind(null, x, y, z);
```

Comment:
`Math.pow(2, z)`, `x * TILE_SIZE` and `y * TILE_SIZE` are repeatedly calculated inside `project`. We can do it once before binding to improve perf.